pkg/util/platform.go

```diff
@@ -13,19 +13,20 @@ const (
 	infrastructureResourceName = "cluster"
 )
 
-func GetPlatform(ctx context.Context, cl client.Reader) (configv1.PlatformType, error) {
-	infra := &configv1.Infrastructure{}
-
-	if err := cl.Get(ctx, client.ObjectKey{Name: infrastructureResourceName}, infra); err != nil {
-		if k8serrors.IsNotFound(err) {
-			return "", nil
-		}
-		return "", err
-	}
-
+func GetPlatform(ctx context.Context, infra configv1.Infrastructure) (configv1.PlatformType, error) {
 	if infra.Status.PlatformStatus == nil {
 		return "", nil
 	}
 
 	return infra.Status.PlatformStatus.Type, nil
 }
+
+func GetInfra(ctx context.Context, cl client.Reader) (configv1.Infrastructure, error) {
+	infra := configv1.Infrastructure{}
```

Comment:
Normally it's a pointer. It's idiomatic in K8s to pass around pointer to your object. It means that it satisfies the interfaces of `client.Object` 
```suggestion
	infra := &configv1.Infrastructure{}
```